### ISSUE TITLE
Matplotlib register_cmap() got an unexpected keyword argument 'lut'

### ISSUE THREAD

**Issue by jcesardasilva** [ptycho/ptypy#318]
I bumped into this problem recently when using Matplotlib 3.4.1:
```
Traceback (most recent call last):
  File "/Users/julio-cesar.da-silva/Documents/python_dev/ptypy/templates/minimal_prep_and_run.py", line 7, in <module>
    from ptypy.core import Ptycho
  File "/usr/local/lib/python3.9/site-packages/ptypy/__init__.py", line 45, in <module>
    from .utils import parallel
  File "/usr/local/lib/python3.9/site-packages/ptypy/utils/__init__.py", line 21, in <module>
    from .plot_utils import *
  File "/usr/local/lib/python3.9/site-packages/ptypy/utils/plot_utils.py", line 416, in <module>
    mpl.cm.register_cmap(cmap=LinearSegmentedColormap(name='franzmap', segmentdata=franzmap_cm), lut=256)
TypeError: register_cmap() got an unexpected keyword argument 'lut'
```

Changing the line 416 in file `ptypy/utils/plot_utils.py`: 
`mpl.cm.register_cmap(cmap=LinearSegmentedColormap(name='franzmap', segmentdata=franzmap_cm), lut=256)`
to:
`mpl.cm.register_cmap(cmap=LinearSegmentedColormap(name='franzmap', segmentdata=franzmap_cm)) #, lut=256)`
Seems to solve the problem, but I don't know if this it is backward compatible. Do we need the `lut` information here? Should I do a quick PR on this or you can handle it directly from your side?

**Comment by daurer** [comment]
@jcesardasilva I think this is a bug that was introduced when this line of code was changed in 7df61de03afa31ee579deb78582d3cff04cf8ece. It should probably be `mpl.cm.register_cmap(cmap=LinearSegmentedColormap(name='franzmap', segmentdata=franzmap_cm, lut=256))` instead. Feel free to make a PR for this.

**Comment by jcesardasilva** [comment]
Thank you @daurer 

In fact, it does not work by doing: 
`pl.cm.register_cmap(cmap=LinearSegmentedColormap(name='franzmap', segmentdata=franzmap_cm, lut=256))`
as you proposed. I now get this: 
```
Traceback (most recent call last):
  File "/Users/julio-cesar.da-silva/Documents/python_dev/ptypy/templates/minimal_prep_and_run_ML_Gaussian.py", line 7, in <module>
    from ptypy.core import Ptycho
  File "/usr/local/lib/python3.9/site-packages/ptypy/__init__.py", line 45, in <module>
    from .utils import parallel
  File "/usr/local/lib/python3.9/site-packages/ptypy/utils/__init__.py", line 21, in <module>
    from .plot_utils import *
  File "/usr/local/lib/python3.9/site-packages/ptypy/utils/plot_utils.py", line 416, in <module>
    mpl.cm.register_cmap(cmap=LinearSegmentedColormap(name='franzmap', segmentdata=franzmap_cm, lut=256))
TypeError: __init__() got an unexpected keyword argument 'lut'
```

I know well the commit you mentioned, since I bumped into similar problems before. But now again, it appears matploblib changed the API again, because it was working like this at the beginning of March.

**Comment by jcesardasilva** [comment]
I have just confirmed that the way it is now, after my old commit  7df61de, works for `Matplotlib==3.3.4` but it begins to break from `Matplotlib>=3.4.0`

**Comment by daurer** [comment]
@jcesardasilva sorry there was a mistake on my part. Can you try it like this: `mpl.cm.register_cmap(cmap=LinearSegmentedColormap(name='franzmap', franzmap_cm, N=256))`. According to the documentation (https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.LinearSegmentedColormap.html) this should work with the latest version of matplotlib, and as far as I can see it should also be backwards compatible to earlier versions.

**Comment by jcesardasilva** [comment]
Thank you again @daurer. Replacing `lut` to `N` seems to have solved the problem indeed. Even by completely removing `N=256` from the game also worked, but there might be a reason to want to keep `N=256` there.

**Comment by daurer** [comment]
Thanks for confirming this @jcesardasilva. Yes, I think the `N=256` should stay there. Maybe you can make a quick PR for this fix?

**Comment by jcesardasilva** [comment]
Closed by the commit #320
